src/messages/MClientReply.h

```diff
@@ -391,27 +388,27 @@ class MClientReply final : public MMDSOp {
     using ceph::decode;
     auto p = payload.cbegin();
     decode(head, p);
+    // for the backward compatability, assuming LINUX error codes on the wire,
+    // errorcode32_t implements conversion from/to different host error codes
+    if (header.version >= HEAD_VERSION) {
+      errorcode32_t temp {static_cast<__s32>(head.result)};
+      temp.convert_decode();
+      head.result = static_cast<__u32>(temp.code);
```

Comment:
i'm a bit confused to understand the to-fro between `__s32` and `__u32`.

Reply:
see comments above regarding signed/unsigned usage for error codes